cmd/argocd/commands/app.go

```diff
@@ -777,7 +776,7 @@ func NewApplicationUnsetCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 			updated := false
 			if app.Spec.Source.Kustomize != nil {
 				for _, paramStr := range parameters {
-					unsetKustomizeOpt(&app.Spec.Source, paramStr);
```

Comment:
I think `argocd app unset` flags we should use consistent with `argocd app set` flags. 
Can you please introduce `--kustomize-image`, `--nameprefix` and `namesuffix` . Flags are defined here: https://github.com/argoproj/argo-cd/blob/ebc048167c2e86f3e119eadb69fd5150ac1f7823/cmd/argocd/commands/app.go#L721

And use `flags.Visit` to iterate through specified list of flags and "unset" corresponding fields. Example it here:

https://github.com/argoproj/argo-cd/blob/ebc048167c2e86f3e119eadb69fd5150ac1f7823/cmd/argocd/commands/app.go#L473-L475

Reply:
Great input. Thank you Alex. Changes are checked-in.